sklearn/tree/tests/test_tree.py

```diff
@@ -1693,17 +1693,28 @@ def test_no_sparse_y_support(name):
 
 
 def test_mae():
-    # check MAE criterion produces correct results
+    # Check MAE criterion produces correct results
     # on small toy dataset
     dt_mae = DecisionTreeRegressor(random_state=0, criterion="mae",
                                    max_leaf_nodes=2)
+
+    # Test MAE where all sample weights are uniform:
+    dt_mae.fit([[3], [5], [3], [8], [5]], [6, 7, 3, 4, 3], np.ones(5))
+    assert_array_equal(dt_mae.tree_.impurity, [1.4, 1.5, 4.0 / 3.0])
+    assert_array_equal(dt_mae.tree_.value.flat, [4, 4.5, 4.0])
+
+    # Test MAE where a `sample_weight` is not explicitly provided.
+    # This is equivalent to providing uniform sample weights, though
+    # the internal logic is different:
     dt_mae.fit([[3], [5], [3], [8], [5]], [6, 7, 3, 4, 3])
-    assert_array_equal(dt_mae.tree_.impurity, [1.4, 1.5, 4.0/3.0])
+    assert_array_equal(dt_mae.tree_.impurity, [1.4, 1.5, 4.0 / 3.0])
     assert_array_equal(dt_mae.tree_.value.flat, [4, 4.5, 4.0])
 
+    # Test MAE where sample weights are non-uniform:
     dt_mae.fit([[3], [5], [3], [8], [5]], [6, 7, 3, 4, 3],
                [0.6, 0.3, 0.1, 1.0, 0.3])
-    assert_array_equal(dt_mae.tree_.impurity, [7.0/2.3, 3.0/0.7, 4.0/1.6])
+    assert_array_almost_equal(dt_mae.tree_.impurity,
+                              [2.5 / 2.3, 0.3 / 0.7, 1.2 / 1.6])
```

Comment:
This illustrates well why I don't love hard coded numerical values: without being an expert in this part of the code, it is hard to see why the new version is correct and not the old one.

Reply:
I agree, unfortunately this file has many other tests with hardcoded values and no explanation.  I was simply following suit.  On a side-note, I have a fix for another issue in this domain with respect to the medians not being calculated correctly.  See https://github.com/scikit-learn/scikit-learn/issues/10725 I will need perhaps more rigorous testing there too rather than one test that works.  It is tricky for me to know best way to integrate as the values can be verified via code at runtime (not a one off test) if desired (as explained in that linked issue)?  

Perhaps you know of test-unit code that I can be re-directed to for how it should really be done properly so that I can follow that instead?

Reply:
Apologies for cross contaminating threads, I only mention this other issue because in most likelihood I will be extending this exact test function!  Ignore this please, I will investigate via other thread.